scipy/stats/_sobol.pyx

```diff
@@ -88,28 +117,22 @@ def _initialize_direction_numbers():
         np.savez_compressed("./_sobol_direction_numbers", vinit=vs, poly=poly)
 
     """
-    cdef cnp.uint64_t[::1] dns_poly
-    cdef cnp.uint64_t[:, :] dns_vinit
-
-    global is_initialized
-    if not is_initialized:
-        dns = np.load(os.path.join(os.path.dirname(__file__),
-                      "_sobol_direction_numbers.npz"))
-        dns_poly = dns["poly"].astype(np.uint64)
-        dns_vinit = dns["vinit"].astype(np.uint64)
-        for i in range(MAXDIM):
-            poly[i] = dns_poly[i]
-        for i in range(MAXDIM):
-            for j in range(MAXDEG):
-                vinit[i][j] = dns_vinit[i, j]
-        is_initialized = True
+    dns = np.load(os.path.join(os.path.dirname(__file__),
+                  "_sobol_direction_numbers.npz"))
+    dns_poly = dns["poly"].astype(dtype)
+    dns_vinit = dns["vinit"].astype(dtype)
+    for i in range(MAXDIM):
+        poly[i] = dns_poly[i]
+    for i in range(MAXDIM):
+        for j in range(MAXDEG):
+            vinit[i][j] = dns_vinit[i, j]
```

Comment:
Probably as efficient to delegate this copying to Numpy, given that you aren't typing `poly`, `dns_poly` etc.

```suggestion
    poly[:MAXDIM] = dns_poly[:MAXDIM]
    vinit[:MAXDIM, :MAXDEG] = dns_vinit[:MAXDIM, :MAXDEG]
```

Reply:
Great, didn't know about that and it's faster.